assets/js/util/index.js

```diff
@@ -590,30 +591,50 @@ export const findTagInHtmlContent = ( html, module ) => {
  * while requesting list of accounts.
  *
  * @param {string} module Module slug.
+ *
+ * @param {string|null} The tag id if found, otherwise null.
  */
 export const getExistingTag = async ( module ) => {
-	try {
-		let tagFound = data.getCache( module + '::existingTag', 300 );
-
-		if ( 'undefined' === typeof tagFound ) {
-			const html = await fetch( `${ googlesitekit.admin.homeURL }?tagverify=1&timestamp=${ Date.now() }` ).then( ( res ) => {
-				return res.text();
-			} );
-
-			tagFound = findTagInHtmlContent( html, module );
-			if ( ! tagFound ) {
-				tagFound = '';
+	const CACHE_KEY = `${ module }::existingTag`;
+	const { homeURL, ampMode } = googlesitekit.admin;
+
+	let tagFound = data.getCache( CACHE_KEY, 300 );
+
+	if ( ! tagFound ) {
+		try {
+			tagFound = await scrapeTag( addQueryArgs( homeURL, { tagverify: 1, timestamp: Date.now() } ), module );
+
+			if ( ! tagFound && 'secondary' === ampMode ) {
+				tagFound = await apiFetch( { path: '/wp/v2/posts?per_page=1' } ).then(
+					// Scrape the first post in AMP mode, if there is one.
+					( posts ) => posts.slice( 0, 1 ).map( async ( post ) => {
+						return await scrapeTag( addQueryArgs( post.link, { amp: 1 } ), module );
+					} ).pop()
+				);
 			}
-		}
 
-		data.setCache( module + '::existingTag', tagFound );
+			// Only set/renew the cache if a tag was found.
+			data.setCache( CACHE_KEY, tagFound || undefined );
```

Comment:
We should cache a falsy value here if no tag was found, but not `undefined` since it essentially means nothing is cached. Since `scrapeTag()` returns `null` in that case, we should cache `null` here too. Note that we would need to actually run the following:
```js
if ( ! tagFound ) {
	tagFound = null;
}
```
That's because if the posts API request doesn't return any posts, `tagFound` will be `undefined` due to return value of `pop()`.

Reply:
The use of `undefined` here was intentional as `setCache` checks for it specifically and does nothing if passed. I've updated it to ensure falsey is always cached as `null`.